code/game/mecha/mech_fabricator.dm

```diff
@@ -204,7 +204,8 @@
 		L.origin_tech = I.origin_tech
 	else
 		I.forceMove(get_step(src, SOUTH))
-	I.materials = final_cost
+	if(istype(I))
```

Comment:
I feel like this entire thing should check if `D.build_path` is an item before creating it. I would dread to think what happens if someone tries to lockbox an IRC frame by accident